pkg/boot/kexec/kexec_load_linux.go

```diff
@@ -65,3 +85,248 @@ func rawLoad(entry uintptr, segments []Segment, flags uint64) error {
 	}
 	return nil
 }
+
+// LoadBzImage loads the given kernel file as to-be-kexeced kernel with
+// the given ramfs file and cmdline string.
+func KexecLoad(kernel, ramfs io.ReaderAt, cmdline string) error {
+	bzimage.Debug = log.Printf
+
+	/* A collection of vars used for processing the kernel for kexec */
+
+	var err error
+	// b is the deserialized bzImage from the kernel
+	// io.ReaderAt.
+	var b bzimage.BzImage
+	// kmem is a struct holding kexec segments.
+	//
+	// It has routines to work with physical memory
+	// ranges.
+	var kmem *Memory
+	// realMode holds setup code.
+	//
+	// If not executing in real mode, it will be just
+	// be setup header.
+	var realMode *LinuxParamHeader
+
+	var kernel16SizeNeeded uint
+
+	var regs32 Entry32Regs
+
+	log.Printf("Start LoadBzImage...")
```

Comment:
I am wondering why you did not use the bzimage package.

Reply:
not sure if I follow, but these are the lines that are using it

(where  `b` is `var b bzimage.BzImage`)
```
	log.Printf("Try decompressing kernel...")
	kb, err := uio.ReadAll(util.TryGzipFilter(kernel))
	if err != nil {
		return err
	}
	log.Printf("Try parsing bzImage...")
	if err := b.UnmarshalBinary(kb); err != nil {
		return err
	}
```

Reply:
probably my confusion. I am going to start looking at this today, to see if I can help. kexec is really a headache.